### PR TITLE
orchagent: Processing port state change notification whenever gPortsOrch is initialized or not

### PR DESCRIPTION
**Pull request by stcheng**
…

### CODE REVIEW

File: orchagent/notifications.cpp [hunk 1]
```diff
@@ -17,15 +17,16 @@ void on_port_state_change(uint32_t count, sai_port_oper_status_notification_t *d
 
     lock_guard<mutex> lock(gDbMutex);
 
-    /* Wait until gPortsOrch is initialized */
-    if (!gPortsOrch || !gPortsOrch->isInitDone())
+    if (!gPortsOrch)
```
Comment (lguohan) [inline orchagent/notifications.cpp]: do we need this?

Comment (lguohan) [inline orchagent/notifications.cpp]: if yes, we still have risk of missing notifications.

Comment (stcheng) [inline orchagent/notifications.cpp]: missing notifications at this stage is acceptable. when the gportsorch pointer is not there, it means that the orchagent does not even starts. during regular orchagent starts, no ports are created before gportsorch pointer is initialized.

Comment (lguohan) [inline orchagent/notifications.cpp]: then, it should be assert since it cannot happen according to what you describe.

Comment (stcheng) [inline orchagent/notifications.cpp]: updated. add SWSS_LOG_WARN level logs.

**Pull request merged**